### PR TITLE
better metadata reading to tokens from being stuck in the bridge

### LINKED ISSUE #222: Suppport more tokens
bytes memory metadata = abi.encode(
            IERC20Metadata(token).name(),
            IERC20Metadata(token).symbol(),
            IERC20Metadata(token).decimals()
        );

Some tokens might not support those methods, one needs to call them using staticcall

### PR DESCRIPTION
**Pull request by josojo**
closes #222

### CODE REVIEW

File: contracts/lib/BridgeAssetOperations.sol [hunk 1]
```diff
@@ -66,10 +65,31 @@ library BridgeAssetOperations {
         if (tokenInfo.originNetwork == 0) {
             revert TokenNotForkable();
         }
+        (bool successNameCall, bytes memory name) = token.staticcall(
+            /// encoding the function signature of N
+            abi.encodeWithSignature("name()")
+        );
+        if (!successNameCall) {
```
Comment (edmundedgar) [inline contracts/lib/BridgeAssetOperations.sol:72]: Is there a reason for our fallbacks to be named subtly differently from the way upstream does it? 

https://github.com/0xPolygonHermez/zkevm-contracts/blob/d70266b8742672d59d4060019538d03fe0aac181/contracts/PolygonZkEVMBridge.sol#L800

**Review by edmundedgar — APPROVED** [review]
Code looks fine, except the comment that polygon upstream seems to be subtly different, it would be good to align it if there isn't a reason not to.

Also I would suggest either adding tests or making an issue or a TODO comment (either in the original code or in the place where the tests should go) that we should add tests.

**Comment by josojo** [comment]
I reused the functions from the bridge contract. It's kinda ucly code duplications, but I can not think of a better method, since I dont' wanna modify the zkevm-contracts repo and refactor the changes there, to keep the diff minimal to the polygon zkevm

**Review by edmundedgar — APPROVED** [review]
Awesome, looks great.

I wouldn't worry about the fact that this copy-pastes some code, I think it's better to copy-paste than do something ugly to make a dependency work.

**Pull request merged**